argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,329 @@
+    detection_fpath: Path
+    ground_truth_fpath: Path
+    figures_fpath: Path
+    detection_cfg: DetectionCfg = DetectionCfg()
+
+    def evaluate(self) -> pd.DataFrame:
+        """Evaluate detection output and return metrics. The multiprocessing
+        library is used for parallel assignment between detections and ground truth
+        annotations.
+
+        Returns:
+            Evaluation metrics of shape (C + 1, K) where C + 1 is the number of classes
+            plus a row for their means. K refers to the number of evaluation metrics.
+        """
+        dt_fpaths = list(self.detection_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
+        gt_fpaths = list(self.ground_truth_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
+
+        assert len(dt_fpaths) == len(gt_fpaths)
+        data: DefaultDict[str, np.ndarray] = defaultdict(list)
+        cls_to_ninst: DefaultDict[str, int] = defaultdict(int)
+
+        with Pool(os.cpu_count()) as p:
+            accum = p.map(self.accumulate, gt_fpaths)
+
+        for frame_stats, frame_cls_to_inst in accum:
+            for cls_name, cls_stats in frame_stats.items():
+                data[cls_name].append(cls_stats)
+            for cls_name, num_inst in frame_cls_to_inst.items():
+                cls_to_ninst[cls_name] += num_inst
+
+        data = defaultdict(np.ndarray, {k: np.vstack(v) for k, v in data.items()})
+
+        init_data = {dt_cls: METRIC_DEFAULT_VALUES for dt_cls in self.detection_cfg.detection_classes}
+        summary = pd.DataFrame.from_dict(init_data, orient="index", columns=METRIC_NAMES)
+        summary_update = pd.DataFrame.from_dict(
+            self.summarize(data, cls_to_ninst), orient="index", columns=METRIC_NAMES
+        )
+
+        summary.update(summary_update)
+        summary = summary.round(self.detection_cfg.significant_digits)
+        summary.index = summary.index.str.title()
+
+        summary.loc["Average Metrics"] = summary.mean().round(self.detection_cfg.significant_digits)
+        return summary
+
+    def accumulate(self, gt_fpath: Path) -> Tuple[DefaultDict[str, np.ndarray], DefaultDict[str, int]]:
+        """Accumulate the statistics for each LiDAR frame.
+
+        Args:
+            gt_fpath: Ground truth file path.
+
+        Returns:
+            cls_stats: Class statistics of shape (N, 8)
+            cls_to_ninst: Mapping of the class names to the number of instances in the ground
+                          truth dataset.
+        """
+        log_id = gt_fpath.parents[1].stem
+        logger.info(f"log_id = {log_id}")
+        ts = gt_fpath.stem.split("_")[-1]
+
+        dt_fpath = self.detection_fpath / f"{log_id}/per_sweep_annotations_amodal/" f"tracked_object_labels_{ts}.json"
+
+        dts = np.array(read_label(str(dt_fpath)))
+        gts = np.array(read_label(str(gt_fpath)))
+
+        cls_stats = defaultdict(list)
+        cls_to_ninst = defaultdict(int)
+        for dt_cls in self.detection_cfg.detection_classes:
```

Comment:
i think might be more readable to name this for loop variable `classname` instead of `dt_cls`, otherwise it seems like it might be detections of a particular class